pkg/handler/v1/provider/openstack.go

```diff
@@ -324,29 +313,29 @@ type OpenstackTenantReq struct {
 
 func DecodeOpenstackTenantReq(_ context.Context, r *http.Request) (interface{}, error) {
 	var req OpenstackTenantReq
-
 	req.Username = r.Header.Get("Username")
 	req.Password = r.Header.Get("Password")
 	req.Domain = r.Header.Get("Domain")
 	req.DatacenterName = r.Header.Get("DatacenterName")
 	req.Credential = r.Header.Get("Credential")
-
 	return req, nil
 }
 
-func getOpenstackCredentials(userInfo *provider.UserInfo, credentialName, username, password, domain, tenant, tenantID string, presetProvider provider.PresetProvider) (string, string, string, string, string, error) {
+func getCredentials(userInfo *provider.UserInfo, cred credential, presetProvider provider.PresetProvider) (credential, error) {
```

Comment:
I would rather do something like:

```
func getPresetCredentials(userInfo *provider.UserInfo, presetName string, presetProvider provider.PresetProvider) (*credentials, error) {
...
```

I would leave the caller to handle the case where a preset name is not given or credentials cannot be obtained. WDYT?

Reply:
Yes that sounds good